edgelet/iotedged/src/lib.rs

```diff
@@ -328,6 +329,18 @@ where
                     crypto.clone(),
                 )?;
 
+                // Normally iotedged will stop all modules when it shuts down. But if it crashed,
+                // then modules will continue to run and they'll be holding stale file descriptors
+                // for the workload and management APIs. On some platforms (e.g. CentOS 7.5), calls
+                // to these APIs will begin to fail. Resilient modules should be able to deal with
+                // this, but we'll restart all modules to ensure a clean start.
+                const STOP_TIME: Duration = Duration::from_secs(30);
```

Comment:
So, in case any module failed to be stopped within the 30 seconds, we'll fail edgelet being started?

Reply:
Correct. 30 seconds is an arbitrary value, so we can adjust that based on feedback. But if we don't fail iotedged in this case, then there are likely to be downstream problems that are more difficult to diagnose, because the errors observed (errors in this zombie module that can't be stopped) will be farther away from the source of the problem.